### PR TITLE
Reflect 3.2.5/6 releases and prepare 3.3.0

### PR DESCRIPTION
**Pull request by autophagy**
## Summary of the changes / Why this improves CrateDB

This PR reflects the release notes of the last 3.2.x releases into the 3.3 branch, and then prepares the release notes for 3.3.0.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/3.2.5.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+- Fixed an issue that could cause queries on the :ref:`sys.snapshots
+  <sys-snapshots>` table to raise an error if a snapshot couldn't be retrieved.
+
+- Fixed an issue that caused inserts into partitioned tables to fail with an
+  ``unknown setting`` error if the table was created in an earlier version of
+  CrateDB using settings that have been removed in later versions.
+
+- Fixed an issue that could result in a ``ConcurrentModificationException``
+  error when querying the ``sys.jobs_metrics`` table.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.2.5.rst]: "while" instead of "when"

File: blackbox/docs/appendices/release-notes/3.2.6.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  in a ``JOIN`` statement.
+
+- Fixed an issue which caused a ``NullPointerException`` when executing a
+  :ref:`window function <window-functions>` over an ordered window which
+  contained null values under the ordered column.
+
+- Fixed an issue which causes sub-select queries with certain ``ORDER BY``
+  constructs to fail.
+
+- Fixed circuit breaker memory accounting of window functions to prevent Out Of
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.2.6.rst]: "Out of Memory" (plus italics)

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Deprecated Settings and Features
+--------------------------------
+
+The query frequency and average duration :ref:`query_stats_mbean` metrics
+has now been deprecated, in favour of the new total count and sum of durations
+metrics.
+
+The :ref:`cluster.graceful_stop.reallocate <cluster.graceful_stop.reallocate>`
+setting has been marked as deprecated. This setting was already being ignored,
+and setting the value to `false` had no effect.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: ``false`` (double ticks)

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+The node decommission using the ``USR2`` :ref:`cli_signals` has now been
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: :ref:`signal <cli_signals>`

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+deprecated in favour of the
+:ref:`ALTER CLUSTER DECOMISSION <alter_cluster_decommission>` statement.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: hard word wrap inside ref text

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+- Added ``current_schemas(boolean)`` scalar function which will return the
+  names of schemas in the ``search_path``.
+
+- Added support for the ``first_value``, ``last_value`` and ``nth_value``
+  window functions as enterprise features.
+
+- Added the ``DROP ANALYZER`` statement to support removal of custom
+  analyzer definitions from the cluster.
+
+- Output the custom analyzer/tokenizer/token_filter/char_filter definition inside
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "analyzer, tokenizer, token_filter, and char_filter definitions"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  the ``information_schema.routines.routine_definition`` column.
+
+- Added support for the window function ``row_number()``.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "``row_number()``  window function"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+- Added support to use any expression in the operand of a ``CASE`` clause.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "support for using any expression"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+PostgreSQL Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Added ``pg_type`` columns: ``typlen``, ``typarray``, ``typnotnull``
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: serial comma after typnotnull

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+  and ``typnamespace`` for improved postgresql compatibility.
+
+- Added a ``pg_description`` table to the ``pg_catalog`` schema for improved
+  postgresql compatibility.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "PostgreSQL"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+- Added support for the window function ``row_number()``.
+
+- Added support to use any expression in the operand of a ``CASE`` clause.
+
+PostgreSQL Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Added ``pg_type`` columns: ``typlen``, ``typarray``, ``typnotnull``
+  and ``typnamespace`` for improved postgresql compatibility.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "PostgreSQL"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+- Added a ``pg_description`` table to the ``pg_catalog`` schema for improved
+  postgresql compatibility.
+
+- Fixed function resolution for postgresql functions ``pg_backend_pid``,
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "PostgreSQL"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+  ``pg_get_expr`` and ``current_database`` when the schema prefix
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: serial comma after pg_get_expr

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+  ``pg_catalog`` is included.
+
+Database Administration
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Added a node check that checks the JVM version under which CrateDB is
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "a node check for the JVM version number." (drop everything after in this sentence)

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+  versions will be dropped in the future.
+
+- Added ``ALTER CLUSTER DECOMMISSION <nodeId | nodeName>`` statement that
+  triggers the existing node decommission functionality.
+
+- Added ``ALTER CLUSTER DECOMMISSION <nodeId | nodeName>`` statement that
+  triggers the existing node decommission functionality.
+
+- Changed the trial license introduced in 3.2 to no longer have an expiration
+  date, but instead be limited to 3 nodes. See :ref:`enterprise_features`.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: "three"

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+
+Deprecations
+~~~~~~~~~~~~
+
+- The query frequency and average duration :ref:`query_stats_mbean` metrics
+  has been deprecated in favour of the new total count and sum of durations
+  metrics.
+
+- Marked the
+  :ref:`cluster.graceful_stop.reallocate <cluster.graceful_stop.reallocate>`
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: hard wrap inside the ref text

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+  setting as deprecated. This setting was already being ignored, setting the
+  value to `false` has no effect.
+
+- The node decommission using the ``USR2`` :ref:`cli_signals` has been
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: :ref:`signal <cli_signals>`

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+  deprecated in favour of the
+  :ref:`ALTER CLUSTER DECOMISSION <alter_cluster_decommission>` statement.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: hard wrap inside the ref text

File: blackbox/docs/appendices/release-notes/3.2.5.rst (hunk 1 above)
Comment (autophagy) [reply blackbox/docs/appendices/release-notes/3.2.5.rst]: I'll make these changes in a seperate PR against master. Little point changing them in this PR.

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+
+Changes
+-------
+
+New Features
+~~~~~~~~~~~~
+
+- Exposed the sum of durations, total, and failed count metrics under the
+  :ref:`query_stats_mbean` for ``QUERY``, ``INSERT``, ``UPDATE``, ``DELETE``,
+  ``MANAGEMENT``, ``DDL`` and ``COPY`` statement types.
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: serial comma after DDL

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+
+- Exposed the sum of statement durations, total, and failed count classified by
+  statement type under the sum_of_durations, total_count and failed_count
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: colum names should be marked up as literals

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+  columns, respectively, in the :ref:`sys-jobs-metrics` table.
+
+SQL Improvements
+~~~~~~~~~~~~~~~~
+
+- Added ``current_schemas(boolean)`` scalar function which will return the
+  names of schemas in the ``search_path``.
+
+- Added support for the ``first_value``, ``last_value`` and ``nth_value``
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: seral comma after last_value

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+  window functions as enterprise features.
+
+- Added the ``DROP ANALYZER`` statement to support removal of custom
+  analyzer definitions from the cluster.
+
+- Output the custom analyzer, tokenizer, token_filter, and char_filter
+  definition inside  the ``information_schema.routines.routine_definition``
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: double space after "inside" should be zapped

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+  column.
+
+- Added support for the ``row_number()`` window function .
```
Comment (mechanomi) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: space before terminal period should be zapped

File: blackbox/docs/appendices/release-notes/3.2.5.rst (hunk 1 above)
Comment (mechanomi) [reply blackbox/docs/appendices/release-notes/3.2.5.rst]: ok. marked as resolved here. added some additional comments. not sure if you want to do them on a second PR also

**Comment by autophagy** [comment]
Retest this please

**Comment by seut** [comment]
@autophagy Damn, my fault, sorry. This will fix it https://github.com/crate/crate/pull/8471

**Comment by autophagy** [comment]
@seut Ah, super! Thank you :sparkles: I'll rebase when it's merged.

**Comment by mfussenegger** [comment]
Please take care that there is a fix with a follow up (https://github.com/crate/crate/pull/8474) that either should be included or will need an additional release notes entry.

**Comment by autophagy** [comment]
@mfussenegger Ok, I will hold off until this is merged, and include it in the release.

**Comment by autophagy** [comment]
Retest this please

File: blackbox/docs/appendices/release-notes/3.3.0.rst [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+
+Database Administration
+~~~~~~~~~~~~~~~~~~~~~~~
+
+- Added a node check for the JVM version number.
+
+- Added ``ALTER CLUSTER DECOMMISSION <nodeId | nodeName>`` statement that
+  triggers the existing node decommission functionality.
+
+- Added ``ALTER CLUSTER DECOMMISSION <nodeId | nodeName>`` statement that
```
Comment (smakalias) [inline blackbox/docs/appendices/release-notes/3.3.0.rst]: dupe?

**Pull request merged**